src/core/server/saved_objects/export/find_with_point_in_time.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Logger } from '../../logging';
+import { SavedObjectsClientContract, SavedObjectsFindOptions } from '../types';
+import { SavedObjectsFindResponse } from '../service';
+
+/**
+ * Returns a generator to help page through large sets of saved objects.
+ *
+ * The generator wraps calls to `SavedObjects.find` and iterates over
+ * multiple pages of results using `_pit` and `search_after`. This will
+ * open a new Point In Time (PIT), and continue paging until a set of
+ * results is received that's smaller than the designated `perPage`.
+ *
+ * Once you have retrieved all of the results you need, it is recommended
+ * to call `close()` to clean up the PIT and prevent Elasticsearch from
+ * consuming resources unnecessarily. This will automatically be done for
+ * you if you reach the last page of results.
+ *
+ * @example
+ * ```ts
+ * const finder = findWithPointInTime({
+ *   logger,
+ *   savedObjectsClient,
+ * });
+ *
+ * const options: SavedObjectsFindOptions = {
+ *   type: 'visualization',
+ *   search: 'foo*',
+ *   perPage: 100,
+ * };
+ *
+ * const responses: SavedObjectFindResponse[] = [];
+ * for await (const response of finder.find(options)) {
+ *   responses.push(...response);
+ *   if (doneSearching) {
+ *     await finder.close();
+ *   }
+ * }
+ * ```
+ */
+export function findWithPointInTime({
```

Comment:
I had split this out into a separate file with the intent of eventually making it public somewhere TBD (there's a [thread above](https://github.com/elastic/kibana/pull/89915#discussion_r573422797) where we are discussing this), but am waiting to change anything else pending the outcome of that discussion.

I agree naming `create...` would make more sense if this remains a static helper, but if we end up exposing it on the SO client (`savedObjectsClient.findWithPointInTime`), then we won't need to use a provider pattern and can simply export the class directly to use in the client. Regardless, I will update this either here or in a follow-up PR once we agree on a plan.

Reply:
For now I've renamed to `createPointInTimeFinder` to make the intent clearer. We decided we will expose this as a helper in a follow-up PR, so it will likely be refactored a bit at that time anyway.